### PR TITLE
Added info --only options

### PR DESCRIPTION
**Pull request by lasote**
Related to #1143

### CODE REVIEW

File: conans/client/command.py [hunk 1]
```diff
@@ -311,7 +311,8 @@ def info(self, *args):
                             'e.g., MyPackage/1.2@user/channel or ./my_project/')
         parser.add_argument("--file", "-f", help="specify conanfile filename")
         parser.add_argument("--only", "-n", nargs="?", const="None",
```
Comment (memsharded) [inline conans/client/command.py]: Missing: update, required, date, author.
And for --path option: export_folder, build_folder, package_folder, source_folder

Comment (memsharded) [inline conans/client/command.py]: I suggest adding a list of accepted options somewhere, and using it both for validation and output.

Comment (lasote) [inline conans/client/command.py]: --paths option is not admitting values:

conan info --paths export_folder

ERROR: Conanfile not found!

Comment (lasote) [inline conans/client/command.py]: Only True or False is allowed in that variable

Comment (memsharded) [inline conans/client/command.py]: Yes, they are options for ``--only``, but only effective when ``--paths``

Comment (lasote) [inline conans/client/command.py]: oh! ok

File: conans/test/command/info_test.py [hunk 2]
```diff
@@ -161,6 +161,14 @@ def only_names_test(self):
         self.assertEqual(["Hello2/0.1@PROJECT", "Hello0/0.1@lasote/stable", "Date",
                           "Hello1/0.1@lasote/stable", "Date"], lines)
 
+        self.client.run("info --only=invalid", ignore_error=True)
+        self.assertIn("Invalid --only value", self.client.user_io.out)
+        self.assertNotIn("with --path specfied, allowed values:", self.client.user_io.out)
```
Comment (memsharded) [inline conans/test/command/info_test.py]: specfied => specified

File: conans/client/command.py [hunk 3]
```diff
@@ -336,6 +345,16 @@ def info(self, *args):
         except:
             reference = os.path.normpath(os.path.join(current_path, args.reference))
 
+        if args.only == ["None"]:
+            args.only = []
+
+        if args.only and args.paths and (set(args.only) - set(path_only_options)):
+            raise ConanException("Invalid --only value '%s' with --path specfied, allowed values: [%s]."
```
Comment (memsharded) [inline conans/client/command.py]: specfied => specified

**Pull request merged**